controllers/ingestion/http/http.go

```diff
@@ -0,0 +1,79 @@
+package http
+
+import (
+	"bytes"
+	"io/ioutil"
+	"net/http"
+)
+
+// DruidHTTP interface
+type DruidHTTP interface {
+	Do() (*Response, error)
+}
+
+// HTTP client
+type Client struct {
+	Method     string
+	URL        string
+	HTTPClient http.Client
+	Body       []byte
+	Auth       Auth
+}
+
+func NewHTTPClient(method, url string, client http.Client, body []byte, auth Auth) DruidHTTP {
+	newClient := &Client{
+		Method:     method,
+		URL:        url,
+		HTTPClient: client,
+		Body:       body,
+		Auth:       auth,
+	}
+
+	return newClient
+}
+
+// Auth mechanisms supported by Druid control plane to authenticate
+// with druid clusters
+type Auth struct {
```

Comment:
If I understand this, `Auth` is not related to the HTTP package. Should be in a different place.  
Also, Why is this `Auth` object exists? it only encapsulates the BasicAuth. Maybe it should be an interface the `BasicAuth` will implement (and in the future other Auth options

Reply:
Auth object can consist of many more mechanisms other than BasicAuth. So in future we can add more support such TLS etc.